app/app.rb

```diff
@@ -78,3 +78,26 @@
   status 400
   body {}
 end
+
+post '/prayers' do
+  email = request.fetch_header('P4L-email')
+  user = User.find_by_email(email)
+
+  resident_id = params.fetch('resident_id')
+  prayer = Prayer.new(resident_id:, user_id: user&.id, recorded_at: Time.current)
+
+  content_type :json
+  return [400, prayer.errors] unless prayer.save!
```

Comment:
I know a better way to do this, but I might wait for the next endpoint to do any refactoring. 🤔